File: src/main/java/org/sopt/common/auth/TokenBlacklistService.java
package org.sopt.common.auth;

import java.time.Duration;

import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Service;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class TokenBlacklistService {

	private final StringRedisTemplate redisTemplate;

	public void blacklist(String accessToken, long remainingSeconds) {
		redisTemplate.opsForValue()
			.set("blacklist:" + accessToken, "logout", Duration.ofSeconds(remainingSeconds));

		// 바로 조회해서 확인
		String value = redisTemplate.opsForValue().get("blacklist:" + accessToken);
		System.out.println("Redis 저장 확인: " + value);
	}

	public boolean isBlacklisted(String accessToken) {
		return Boolean.TRUE.equals(redisTemplate.hasKey("blacklist:" + accessToken));
	}
}

File: src/main/kotlin/org/sopt/auth/TokenBlacklistService.kt
package org.sopt.auth

import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.stereotype.Service
import java.time.Duration

@Service
class TokenBlacklistService(
    private val redisTemplate: StringRedisTemplate
) {
    fun blacklist(
        accessToken: String,
        remainingSeconds: Long
    ) {
        redisTemplate.opsForValue().set(
            "blacklist:$accessToken",
            "logout",
            Duration.ofSeconds(remainingSeconds)
        )
    }

    fun isBlacklisted(accessToken: String): Boolean =
        redisTemplate.hasKey("blacklist:$accessToken") == true
}
